include/vecmath/vec.h

```diff
@@ -402,12 +402,12 @@ namespace vm {
                 return -1;
             } else if (lhsIsNaN && !rhsIsNaN) {
                 return 1;
-            }
-
-            if (lhs[i] < rhs[i] - epsilon) {
-                return -1;
-            } else if (lhs[i] > rhs[i] + epsilon) {
-                return 1;
+            } else if (!lhsIsNaN && !rhsIsNaN) {
```

Comment:
This is a little clearer - the behaviour doesn't change though, so the commit message is wrong - the code did the same thing before because both of these `if` conditions fail when lhs[i] and rhs[i] are both nan:

```
if (lhs[i] < rhs[i] - epsilon) {
                    return -1;
                } else if (lhs[i] > rhs[i] + epsilon) {
                    return 1;
                }
```

Reply:
It's a similar problem as before. Clang does not allow floating point math in constexpr contexts if they involve NaN, it seems. With this change, we prove to the compiler that We really aren't doing any maths unless we know the operands are not NaN. I'll fix the commit message to make this clearer.